src/node-file-trace.ts

```diff
@@ -193,50 +199,32 @@ export class Job {
     return false;
   }
 
-  async stat (path: string) {
-    const cached = this.statCache.get(path);
-    if (cached) return cached;
-    await this.fileIOQueue.acquire();
+  protected async statFn (path: string) {
     try {
-      const stats = await fsStat(path);
-      this.statCache.set(path, stats);
-      return stats;
+      return await fsStat(path);
     }
     catch (e: any) {
       if (e.code === 'ENOENT') {
-        this.statCache.set(path, null);
```

Comment:
What about deleting from the cache?

The new wrapper doesn't do that 🤔

Reply:
The wrapper does cache the surrounding Promise, so in case of an error and when `null` is returned it will also be part if the cached promise.

As far as I understood how the cache works it doesn't do cache invalidation anyway. Once cached it is cached forever. :)